### PR TITLE
LPS-130658 | Add sample test for frontend js state

### PR DESCRIPTION
**Pull request by john-co**
Resubmitting from feedback to revise and match portal project sdk template in https://github.com/brianchandotcom/liferay-portal/pull/101764#issuecomment-847353876

**Description**
Adds a sample portlet to test the acceptance criteria for js global state epic in https://issues.liferay.com/browse/LPS-122065

**Test Plan**
1. Clicking on increment JSP button adds +1 to js state and is instantaneously reflected on JSP Counter
2. Clicking on increment react button adds +1 to js state is instantaneously reflected on the React app counter
3. Typing in the form field will change the card to show the text and string length
4. Typing in the form field will show the same text in JSP (right side) 

https://user-images.githubusercontent.com/35235266/117898116-08998d80-b279-11eb-866b-cf2d6f992e67.mp4

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-130658

### CODE REVIEW

**Comment by john-co** [comment]
ci:test:sf

File: modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+javascriptsample.caption=Hello from JavaScript Sample!
```
Comment (nhpatt) [inline modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/content/Language.properties:1]: I think this should be javascript-sample-caption. Are you using it?

**Review by nhpatt — APPROVED** [review]
I've made a very quick review and it looks good, nothing eye-soring. I'll review it a bit more in depth in a bit :)

File: modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/content/Language.properties (hunk 1 above)
Comment (john-co) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/content/Language.properties:1]: Thanks, and I don't think I'm using the variable...I'll check rq. Will remove if not being used.

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
PortalSmoke passed in `ci:test:stable` so the unique failure looks like a fluke. Looking at the poshi report in detail, it doesn't look related to the commits.

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
Rerunning since I forgot to manually forward it's been almost a week...will manually forward if looks ok.

File: modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
 		<button id="test-button-react">Increment react</button>
 
-		<aui:script require="@liferay/frontend-js-state-web@1.0.3/index as StateModule, @liferay/frontend-js-sample-web@1.0.0/js/sharedState as SharedState">
+		<aui:script require="@liferay/frontend-js-state-web@1.0.6/index as StateModule, @liferay/frontend-js-sample-web@1.0.0/js/sharedState as SharedState">
```
Comment (diegonvs) [inline modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: Hey John, about `@liferay/frontend-js-state-web@1.0.6`, did you already tried to use `@liferay/frontend-js-state-web`?

Comment (john-co) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: Testing that out right now, your suggestion makes me feel better about my guess on the syntax :)

Comment (john-co) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: It didn't work... :(

Using 
```
		<aui:script require="@liferay/frontend-js-state-web/index as StateModule, @liferay/frontend-js-sample-web/js/sharedState as SharedState">
```

There are console errors related to those dependencies
Selection_077

And the jsp portion is not displayed properly after publishing the portlet
Selection_078

Comment (john-co) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: Hi @izaera , any idea on how to troubleshoot this? Is there a syntax or step that I may be missing to generalize the version dependency?

Comment (nhpatt) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: Hi John! We've discussing this issue (with another module) here: https://liferay.slack.com/archives/C3JBR21HA/p1628698250243600

TL;DR: there is a PR that should fix it.

Comment (john-co) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: Thanks!

Note: depends on PR https://github.com/liferay-frontend/liferay-portal/pull/1367#issuecomment-901456661

File: modules/apps/frontend-js/frontend-js-sample-web/.npmbundlerrc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (izaera) [inline modules/apps/frontend-js/frontend-js-sample-web/.npmbundlerrc:1]: This should go in `npmscripts.config.js`, since we removed all `.npmbundlerrc` files long ago, moving their contents there.

File: modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp (hunk 2 above)
Comment (izaera) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: It should work without the version numbers. `<aui:script require>` has support for that, as long as you declare the required packages in your project's `package.json` and use the bundler to build it...

Comment (izaera) [reply modules/apps/frontend-js/frontend-js-sample-web/src/main/resources/META-INF/resources/view.jsp:37]: Look for `<aui:script require` inside portal's sourcetree and you'll see that nobody specifies the version number.

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:test:sf

File: modules/apps/frontend-js/frontend-js-sample-web/.npmbundlerrc (hunk 3 above)
Comment (john-co) [reply modules/apps/frontend-js/frontend-js-sample-web/.npmbundlerrc]: Thanks! moved the contents to npmscripts.config.js

**Comment by john-co** [comment]
After removing the strict dependency, I'm still not able to see the react portion (left side).

Selection_009

Is there something else I'm missing? cc/ @izaera @bryceosterhaus

**Comment by john-co** [comment]
Will close this for now since this prob won't be fixed soon: https://liferay.slack.com/archives/C3JBR21HA/p1631539061087800?thread_ts=1631053192.040200&cid=C3JBR21HA

**Comment by izaera** [comment]
I'm going to have a look at this so that I can unblock it (and maybe https://github.com/liferay-frontend/liferay-portal/pull/1510 too)

**Comment by izaera** [comment]
@bryceosterhaus :point_up:

**Comment by izaera** [comment]
Fixed at https://github.com/liferay-frontend/liferay-portal/pull/1511

Note that I fixed the PR, but didn't use `<aui:script require>` with external modules (see https://github.com/liferay-frontend/liferay-portal/pull/1511#issue-745657846)
